R/give_opinions.R

```diff
@@ -1,127 +1,169 @@
-    readme_issues <- give_opinions_readme(readme,
-                                          desc_info$identifier)
-  }
+give_opinions <- function(pkg_path = getwd()) {
 
+  # set the path to the description file
+  description_file <- file.path(pkg_path, "DESCRIPTION")
 
-  fixmes <- rbind(descr_issues, readme_issues)
+  # read the description file
+  description <- read_description_if_null(NULL, description_file)
 
-  if(!is.null(fixmes)){
-    descr <- desc::desc(descr_path)
-    fixmes$package <- as.character(descr$get("Package"))
-    return(fixmes)
-  }else{
-    return(NULL)
+  # create a list of issues, each element is a vector of character or NULL
+  fixmes <- do.call(rbind, list(
+
+    # opinions about DESCRIPTION
+    give_opinions_desc(description = description),
+
+    # opinions about README
+    try_to_give_opinions_readme(description_file)
+  ))
+
+  if (! is.null(fixmes)) {
+
+    fixmes$package <- as.character(description$get("Package"))
   }
 
+  fixmes
+}
+
+# read_description_if_null -----------------------------------------------------
+read_description_if_null <- function(description, description_file) {
+
+  if (is.null(description)) {
+
+    desc::desc(description_file)
+
+  } else {
 
+    description
+  }
 }
 
-give_opinions_desc <- function(descr_path){
-  descr <- desc::desc(descr_path)
+# give_opinions_desc -----------------------------------------------------------
+give_opinions_desc <- function(description_file, description = NULL) {
+
+  # read description from description_file if description is NULL
+  description <- read_description_if_null(description, description_file)
+
+  # Start with an empty vector of "fixme" messages
+  fixmes <- character()
+
   # Authors
-  if (inherits( try(descr$get_authors(), silent = TRUE),
-                'try-error')){
-    authors_fixme <- "The syntax Authors@R instead of plain Author and Maintainer is recommended in particular because it allows richer metadata" # no lint
-  }else{
-    authors_fixme <- NULL
+  if (fails(description$get_authors())) {
+
+    fixmes <- c(fixmes, get_message(message_id = "hint_use_authors_r"))
   }
 
   # URL
-  if(is.na(descr$get("URL"))){
-    url_fixme <- "URL field. Indicate the URL to your code repository."
-  }else{
-    checkurls <- check_urls(descr$descr$get_urls)
-    if(checkurls != ""){
-      url_fixme <- checkurls
-    }else{
-      url_fixme <- NULL
-    }
+  fixmes <- add_url_fixmes(
+    fixmes = fixmes,
+    main_url = description$get("URL"),
+    message_id = "hint_add_repo_url",
+    further_urls = description$descr$get_urls
+  )
+
+  # BugReports
+  fixmes <- add_url_fixmes(
+    fixmes = fixmes,
+    main_url = description$get("BugReports"),
+    message_id = "hint_add_bug_report_url"
+  )
+
+  fixmes_as_tibble_or_message(fixmes, where = "DESCRIPTION")
+}
+
+# fails ------------------------------------------------------------------------
+fails <- function(expr, silent = TRUE) {
```

Comment:
I like this function a lot! Could it be moved to utils.R?

Reply:
Bien sûr!